src/zen/edgescroll/ZenEdgeScrollManager.mjs

```diff
@@ -0,0 +1,340 @@
+{
+  const EDGE_INTERACTION_WIDTH_PX = Services.prefs.getIntPref('zen.theme.border-radius', 8);
+  const SYNTHETIC_EVENT_X_OFFSET_FROM_RIGHT_EDGE = 2;
+  const ACTOR_NAME = 'ZenEdgeScroll'; // Name used for actor registration
+
+  class ZenEdgeScrollManager extends ZenDOMOperatedFeature {
+    init() {
+      this.isSynthesizingDrag = false;
+      this.dragInitialModel = {
+        targetBrowserDuringDrag: null,
+        targetBrowsingContextDuringDrag: null,
+      };
+      this.edgeScrollTriggerDiv = null; // Added for the trigger div
+
+      this._boundHandleMouseDown = this.handleMouseDown.bind(this);
+      this._boundHandleSyntheticDrag = this.handleSyntheticDrag.bind(this);
+      this._boundHandleSyntheticDragEnd = this.handleSyntheticDragEnd.bind(this);
+      this._boundHandleWheel = this.handleWheel.bind(this);
+      this._boundUpdateTriggerDivDisplay = this._updateTriggerDivDisplay.bind(this); // Added
+      this._initialized = false;
+
+      if (window.gZenEdgeScrollManager._initialized === true) {
+        console.warn('ZenEdgeScrollManager is already initialized.');
+        return;
+      }
+      window.gZenEdgeScrollManager._initialized = true;
+
+      // Create and append the edge scroll trigger div
+      this.edgeScrollTriggerDiv = window.document.createElement('div');
+      this.edgeScrollTriggerDiv.id = 'zen-edge-scroll-trigger';
+      Object.assign(this.edgeScrollTriggerDiv.style, {
+        position: 'fixed',
+        top: '0px',
+        right: '0px',
+        width: `${EDGE_INTERACTION_WIDTH_PX}px`,
```

Comment:
If migrated to the CSS file, please use `ver(--zen-element-separation)` instead of the border radius preference.